### PR TITLE
Upgrade to Zipkin V2

### LINKED ISSUE #5: Upgrade to Zipkin V2

### PR DESCRIPTION
**Pull request by sfackler**
There are quite a few API changes here:

* Binary annotations are gone.
* Annotations no longer contain endpoints.
* Annotations are now constructed directly via `new` and `now` rather
    than using a builder.
* A Span contains new fields:
    * `kind` - An enum identifying the kind of span. This replaces the
        "special" `lc`, `cs`, `sr` etc annotations.
    * `shared` - Identifies if this span is joining an existing one
        created on a client. Server spans set this rather than omitting
        a timestamp and duration.
    * `localEndpoint` - The service's endpoint.
    * `remoteEndpoint` - The peer's endpoint.
    * `tags` - A String -> String map replacing binary annotations.
* Endpoint's `serviceName` and `Span`'s `name` are now optional.
* OpenSpan has a `name` setter, and that argument has been removed from
    the various `Tracer` methods.
* OpenSpan's `server` and `client` methods have been removed in favor of
    the `kind` method.

Closes #5

### CODE REVIEW

File: zipkin/src/lib.rs [hunk 1]
```diff
@@ -98,6 +94,7 @@ where
     S: Serializer,
 {
     let micros = duration.as_secs() * 1_000_000 + duration.subsec_nanos() as u64 / 1_000;
+    let micros = micros.max(1);
```
Comment (bavardage) [inline zipkin/src/lib.rs]: just a convenience or now also part of zipkin v2?

File: zipkin/src/test.rs [hunk 2]
```diff
@@ -16,17 +16,17 @@ fn detach() {
...
         .reporter(Box::new(TestReporter(spans.clone())))
-        .build(Endpoint::builder().build("test"));
+        .build(Endpoint::builder().build());
 
-    let parent = tracer.new_trace("parent");
+    let parent = tracer.new_trace();
     let parent_trace = parent.context().trace_id();
     let parent_id = parent.context().span_id();
 
-    let mut detached = tracer.next_span("detached");
```
Comment (bavardage) [inline zipkin/src/test.rs]: spans still have names though right? worth having some way of setting the name on the `OpenSpan`?

Comment (sfackler) [reply zipkin/src/test.rs]: Oops yeah I thought I already had that.

File: zipkin/src/lib.rs (hunk 1 above)
Comment (sfackler) [reply zipkin/src/lib.rs]: It's part of the spec: https://github.com/openzipkin/zipkin-api/blob/master/zipkin2-api.yaml#L369-L371

**Comment by sfackler** [comment]
Yeah, you can emit a V2 span as a V1 span. Here's the Java library's implementation of that for example: https://github.com/openzipkin/zipkin/blob/master/zipkin2/src/main/java/zipkin2/internal/V1SpanWriter.java

**Pull request merged**